### PR TITLE
[SYCL] CMake/LIT support for in-tree end-to-end tests

### PR DESCRIPTION
**Pull request by aelovikov-intel**
Together with a few of existing tests to be able to test changes in infrastructure to use them (both Github CI in a future PR and internally).

For the final move we plan to erase the entire subdirectory and merge a new PR without "squash" to preserve history.

### CODE REVIEW

File: sycl/test-e2e/README.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+   but can be built in compiler project (e.g. with "ninja check").
+ - Target runtime(s) to execute tests on devices. See
+   [installation instructions](https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#install-low-level-runtime)
+
+# Build and run tests
+
+Get sources
+
+```
+git clone https://github.com/intel/llvm-test-suite
```
Comment (bader) [inline sycl/test-e2e/README.md:30]:
```suggestion
git clone https://github.com/intel/llvm
```
?

**Review by bader — COMMENTED** [review]
> Together with a few of existing tests to be able to test changes in infrastructure to use them (both Github CI in a future PR and internally).

I think GitHub CI part is missing.

**Comment by aelovikov-intel** [comment]
Only last two commits have meaningful changes, the rest is coming from the https://github.com/intel/llvm-test-suite via `git-filter-repo`. This new code is essentially dead at this point and will be tested once add the CI support in a separate PR or through tests in internal testing infrastructure.

Can either be merged directly or squash-and-merge'd.

**Review by v-klochkov — COMMENTED** [review]
Ok from ESIMD component. Thank you!.

File: sycl/test-e2e/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    )
+    set_target_properties(${TARGET} PROPERTIES FOLDER "SYCL Level Zero tests")
+    add_dependencies(check-sycl-e2e ${TARGET})
+
+  endforeach()
+endif(SYCL_TEST_E2E_TARGETS)
+
+# TODO: Re-enable when moving the entire intel/llvm-test-suite back in-tree.
+# add_subdirectory(External)
+# add_subdirectory(ExtraTests)
```
Comment (againull) [inline sycl/test-e2e/CMakeLists.txt:72]: Should we reenable this or is this not relevant anymore?

Comment (aelovikov-intel) [reply sycl/test-e2e/CMakeLists.txt:72]: We don't move these directories yet, so they cannot be uncommented yet.

Comment (againull) [reply sycl/test-e2e/CMakeLists.txt:72]: Got it, thanks!

**Review by v-klochkov — APPROVED** [review]

**Pull request merged**